src/mono/mono/metadata/metadata.c

```diff
@@ -6397,14 +6390,13 @@ mono_metadata_methods_from_event   (MonoImage *meta, guint32 index, guint *end_i
 			break;
 	}
 	end = start + 1;
-	int rows = mono_metadata_table_num_rows (meta, MONO_TABLE_METHODSEMANTICS);
-	while (end < rows) {
+	while (end < mono_metadata_table_num_rows (meta, MONO_TABLE_METHODSEMANTICS)) {
```

Comment:
Don't do this. We don't want to call `mono_metadata_table_num_rows` on each loop iteration.